.github/workflows/lint.yml

```
@@ -15,7 +15,7 @@ jobs:
       # NOTE: We intentionally lint against our minimum supported Python.
       - uses: actions/setup-python@61a6322f88396a6271a6ee3565807d608ecaddd1
         with:
-          python-version: "3.7"
+          python-version: "3.8"
```

I think we can use `python-version-file` here now and elsewhere:

```suggestion
          python-version-file: "pyproject.toml"
```

That should reduce the number of locations we need to update these going forwards 🙂 

Ref: https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#using-the-python-version-file-input

---

Noting for future reference: this causes us to resolve the latest compatible version instead, which will generally be wrong in terms of what we want with linting (we want the earliest compatible version).